### PR TITLE
ci: use micromamba and uv and also install ROOT

### PR DESCRIPTION
**Pull request by ikrommyd**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,21 +30,23 @@ jobs:
     name: JIT compiled tests for Python ${{ matrix.python-version }}
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5.0.0
-        name: Set up Python ${{ matrix.python-version }}
-
+      - uses: mamba-org/setup-micromamba@v1
+        name: Set up Micromamba environment
```
Comment (matthewfeickert) [inline .github/workflows/ci.yml:41]: @ikrommyd This is a drive by comment but I would use `mamba-org/setup-micromamba@v1` like you're doing now and then just specify the Python and ROOT version you want like: https://github.com/mamba-org/setup-micromamba?tab=readme-ov-file#create-environment-from-environment-specs

remember to setup the shell defaults though

```yaml
    defaults:
      run:
        # The login shell is necessary for the setup-micromamba setup
        # to work in subsequent jobs.
        # https://github.com/mamba-org/setup-micromamba#about-login-shells
        shell: bash -e -l {0}
```

Comment (ikrommyd) [reply .github/workflows/ci.yml:41]: I think I did it. Thanks!. I basically want to test against some roofit PDFs so ROOT version shouldn't really matter. I think I can leave it unpinned for now.

Comment (ikrommyd) [reply .github/workflows/ci.yml:41]: The only thing I don't like right now is this repetitive use of `shell: micromamba-shell {0}`. I haven't found out if I can just activate the environment and keep it active somehow.

Comment (matthewfeickert) [reply .github/workflows/ci.yml:41]: @ikrommyd c.f. my comment above in https://github.com/zfit/zfit-physics/pull/69#discussion_r1542178078 that shows how to do this for the entire job (also shown in https://github.com/mamba-org/setup-micromamba?tab=readme-ov-file#about-login-shells and a few other places)

Comment (ikrommyd) [reply .github/workflows/ci.yml:41]: ah I'm stupid. Yeah thanks got it

**Review by matthewfeickert — COMMENTED** [review]
LGTM now.

**Comment by ikrommyd** [comment]
@jonas-eschle please take a look and let me know if you're good with that change. I need it to write tests for https://github.com/zfit/zfit-physics/pull/68 and may also make sense to add tests against ROOT for the other distributions we got. May also be a good change to do for zfit.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -56,14 +66,18 @@ jobs:
...
         with:
-          python-version: ${{ matrix.python-version }}
-      - name: Install dependencies
+          environment-name: test-env
+          create-args: >-
+            python=${{ matrix.python-version }}
+            pip
```
Comment (jonas-eschle) [inline .github/workflows/ci.yml:76]: We can use 'uv', familiar with it? Just install uv and use it as 'uv pip install'
Is significantly faster, quite new

**Review by jonas-eschle — APPROVED** [review]
Makes sense, maybe switch to uv? But if that doesn't work out of the box, standard pip isn't wrong either, of course. Up to you, merge when your satisfied

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ikrommyd) [reply .github/workflows/ci.yml:76]: Yeah, I'm using it locally. It's doable but @matthewfeickert, wasn't there an issue with tensorflow installation? I remember you brought something up at some point on slack

Comment (ikrommyd) [reply .github/workflows/ci.yml:76]: @jonas-eschle https://github.com/astral-sh/uv/issues/2062, https://github.com/scikit-hep/pyhf/pull/2444

Comment (ikrommyd) [reply .github/workflows/ci.yml:76]: Ok `uv` needs to install in editable mode to install from git. That's stupid

Comment (matthewfeickert) [reply .github/workflows/ci.yml:76]: `uv` is great, and I'm working to get the `pyhf` CI to use it, but I agree with @ikrommyd that is a whole PR unto itself and atomicly different from this one.

I think you can get ahead of `pyhf` when it comes to using `uv` in CI, but I'll keep you updated as well. :)

**Pull request merged**